bookstore/client/nb_client.py

```diff
@@ -126,30 +73,101 @@ def get_contents(self, path):
         return resp.json()
 
 
-def python_compat_session(session):
+class LiveNotebookServerRecord(NamedTuple):
+    """Representation of a live notebook server.
+
+    This NamedTuple is an immutable representation of the object returned by
+    ``notebook.notebookapp.list_running_servers()``.
+    """
+
+    base_url: str
+    hostname: str
+    notebook_dir: str
+    password: bool
+    pid: int
+    port: int
+    secure: bool
+    token: str
+    url: str
+
+
+def extract_kernel_id(connection_file):
+    """Helper to get a kernel id number from a filename"""
+    return os.path.basename(connection_file).lstrip('kernel-').rstrip('.json')
+
+
+class KernelInfo:
+    """Representation of Kernel information for a running kernel
+
+       Example
+       -------
+       kernel_id: str  # 'f92b7c8b-0858-4d10-903c-b0631540fb36',
+       name: str  # 'dev',
+       last_activity: str  #'2019-03-14T23:38:08.137987Z',
+       execution_state: str  #'idle',
+       connections: int  # 0 connections to the kernel
+    """
+
+    def __init__(self, kernel_id, name, last_activity, execution_state, connections):
+        # TODO: verify that *args is not needed
+        self.kernel_id = kernel_id
+        self.name = name
+        self.last_activity = last_activity
+        self.execution_state = execution_state
+        self.connections = connections
+
+
+def python_compatible_session(session):
+    """Helper to make a deep copy of a session"""
     deepcopy(session)
 
 
+class NotebookSession:
+    """Representation of an active session between a notebook and a kernel
+
+       Example
+       -------
+       id: str  # '68d9c58f-c57d-4133-8b41-5ec2731b268d',
+       path: str  # 'Untitled38.ipynb',
+       name: str  # '',
+       type: str  # 'notebook',
+       kernel: KernelInfo
+       notebook: dict  # {'path': 'Untitled38.ipynb', 'name': ''}
+    """
+
+    def __init__(self, path, name, session_type, kernel, notebook):
+        # TODO: double check *args, **kwargs are not necessary
+        self.path = path
+        self.name = name
+        self.type = session_type
+        self.kernel = KernelInfo(**kernel)
+        self.notebook = notebook
+
+
 class NotebookClientCollection:
+    """Representation of a collection of notebook clients"""
+
+    # TODO: refactor from lambda to a def
     nb_client_gen = lambda: (NotebookClient(x) for x in list_running_servers())
     sessions = {x.url: x.sessions for x in nb_client_gen()}
 
     @classmethod
     def current_server(cls):
+        """class method for current notebook server"""
         for server_url, session_dict in cls.sessions.items():
             for session_id, session in session_dict.items():
-                python_compat_session(session)
-                if NotebookSession(**session).kernel.id == extract_kernel_id(
+                python_compatible_session(session)
+                if NotebookSession(**session).kernel.kernel_id == extract_kernel_id(
```

Comment:
I feel mixed about the idea that we're having the accessor being different than the spec's response. I totally agree that it's clearer though from the perspective of distinguishing from the session's id.